general/cisetup/pipelinesyaml.go

```diff
@@ -119,29 +119,22 @@ func (yg *JFrogPipelinesYamlGenerator) createPipeline(pipelineName, gitResourceN
 	}, nil
 }
 
-func (yg *JFrogPipelinesYamlGenerator) createSteps(gitResourceName, buildInfoResourceName string) ([]PipelineStep, error) {
-	var steps []PipelineStep
-	previousStepName := ""
-	for tech := range yg.SetupData.BuiltTechnologies {
-		switch tech {
-		case Maven:
-			curStep := yg.createMavenStep(gitResourceName, previousStepName)
-			steps = append(steps, curStep)
-			previousStepName = curStep.Name
-		case Gradle:
-			curStep := yg.createGradleStep(gitResourceName, previousStepName)
-			steps = append(steps, curStep)
-			previousStepName = curStep.Name
-		case Npm:
-			curStep, err := yg.createNpmStep(gitResourceName, previousStepName)
-			if err != nil {
-				return nil, err
-			}
-			steps = append(steps, curStep)
-			previousStepName = curStep.Name
+func (yg *JFrogPipelinesYamlGenerator) createSteps(gitResourceName, buildInfoResourceName string) (steps []PipelineStep, err error) {
+	var step PipelineStep
+
+	switch yg.SetupData.BuiltTechnology.Type {
+	case Maven:
+		step = yg.createMavenStep(gitResourceName, "")
```

Comment:
The input of `previousStepName` and the flow that uses it can also be removed from all these three steps. The flow checks if the previous name is not empty, but it always is.
Sorry for not spotting this earlier.